client/src/App.js

```diff
@@ -0,0 +1,101 @@
+import React, { Component } from 'react';
+import Dashboard from './components/Dashboard';
+import Display from './components/Display';
+import './App.css';
+
+class App extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            strikes: 0,
+            balls: 0,
+            inning: 0,
+            inningsData: [
+                {outs: 0, runs: 0, errors: 0},
+                {outs: 0, runs: 0, errors: 0},
+                {outs: 0, runs: 0, errors: 0},
+                {outs: 0, runs: 0, errors: 0},
+                {outs: 0, runs: 0, errors: 0},
+                {outs: 0, runs: 0, errors: 0},
+                {outs: 0, runs: 0, errors: 0},
+                {outs: 0, runs: 0, errors: 0},
+                {outs: 0, runs: 0, errors: 0}
+            ],
+            basesOccupied: {
+                base1: false,
+                base2: false,
+                base3: false
+            }
+        };
+
+        this.handleBall = this.handleBall.bind(this);
+        this.handleFoul = this.handleFoul.bind(this);
+        this.handleStrike = this.handleStrike.bind(this);
+        this.handleHit = this.handleHit.bind(this);
+        this.resetCounters = this.resetCounters.bind(this);
+    }
+
+    handleBall(){
+        if(this.state.balls === 3){
+            this.resetCounters();
+        }else{
+            this.setState({
```

Comment:
```suggestion
            this.setState(prevState => ({
                ...prevState, 
                balls: ++prevState.balls
            }))
```
This would be the better approach as it doesn't mutate state directly